libyul/Object.cpp

```diff
@@ -65,9 +65,45 @@ set<YulString> Object::dataNames() const
 {
 	set<YulString> names;
 	names.insert(name);
-	for (auto const& subObject: subIndexByName)
-		names.insert(subObject.first);
-	// The empty name is not valid
+	for (auto const& subObjectNode: subObjects)
+	{
+		names.insert(subObjectNode->name);
+		if (auto const* subObject = dynamic_cast<Object const*>(subObjectNode.get()))
+			for (YulString const& subSubObj: subObject->dataNames())
+				names.insert(YulString{subObject->name.str() + "." + subSubObj.str()});
+	}
+
 	names.erase(YulString{});
 	return names;
 }
+
+map<YulString, shared_ptr<ObjectNode>> Object::subObjectsByDataName() const
+{
+	map<YulString, shared_ptr<ObjectNode>> objectsByDataName;
```

Comment:
I think this should really have the `sub` prefix for clarity. Perhaps change the name a little so that it does not conflict with function name?
```suggestion
	map<YulString, shared_ptr<ObjectNode>> subObjectMap;
```